packages/expo-updates-interface/android/src/main/java/expo/modules/updatesinterface/UpdatesInterface.java

```diff
@@ -0,0 +1,34 @@
+package expo.modules.updatesinterface;
+
+import android.content.Context;
+
+import org.json.JSONObject;
+
+import java.util.HashMap;
+
+/**
+ * Interface for other modules that depend on expo-updates for loading production updates but do not
+ * want to depend on expo-updates or delegate control to the singleton UpdatesController.
+ */
+public interface UpdatesInterface {
+
+  interface UpdateCallback {
+    void onFailure(Exception e);
+    void onSuccess(Update update);
+    void onProgress(int successfulAssetCount, int failedAssetCount, int totalAssetCount);
+
+    /**
+     * Called when a manifest has been downloaded. The return value indicates whether or not to
+     * continue to download the update described by this manifest. Returning `false` will abort the
+     * load and no other callback methods will be called.
+     */
+    boolean onManifestLoaded(JSONObject manifest);
```

Comment:
Do we want this to use the Manifest types that @wschurman defined? (Related: should the Manifest type definitions move into this interface package?)

Reply:
That's a good point; probably so to both questions. Currently the dev-launcher defines its own manifest types, but we'll probably want to move to @wschurman 's classes anyway in order to more easily move over to the new manifest format, once we're adding EAS update support.

@lukmccall any thoughts or objections to that? Otherwise I'll work on adding the types to this package in preparation for that.

Reply:
I think it's a good idea 👍

Reply:
Ok, going to land this now, along with #13031 and #13032, and follow up with a separate PR since that will touch all 3 places again.